docs/30DaysOfPWA/advanced-capabilities/06.md

```diff
@@ -1,15 +1,174 @@
+  if (event.tag === 'update-feed-content') {
+      event.waitUntil(
+
+        // You have to implement the function that fetches the latest posts and updates the storage:
+        updateFeedContent()
+      );
+  }
+});
+```
+
+Please note, that it's not an obligation but a possibility for you to fetch new data in `updateFeedContent` function. Before sending a server request, you might want to check if the user is on a non-metered connection and if there is enough space in the browser storage using Network Information and StorageManager APIs respectively.
+
+## Privacy and resource utilization considerations
+
+Both Background Sync and Periodic Background Sync APIs are designed to run some custom code (that might send network requests) in the background, at any time, and without notifying a user. It raises at least two concerns: privacy and resource utilization. Any network request is potentially a history leaking and location tracking reason. And any code consumes CPU, memory, battery, and, if needed, network resources. To avoid these problems, both specifications have requirements and recommendations that browser vendors must follow while implementing these APIs. They include:
+
+- Having Background Sync permission granted for the origin. It has `allow` state by default but users can deny it.
+
+![Background Sync permission](_media/background-sync-permission.png)
+
+- Capping the number of background sync retries and duration of periodic syncs. It's not defined in API specification, but for the periodic sync intervals, browsers use the _Site Engagement_ index (number and duration of user interactions) for the particular origin on a particular device to calculate the exact intervals. At best it is a few hours. Hint: you can check the Site Engagement index for yourself on your current Chromium-based browser by visiting [about://site-engagement/](about://site-engagement/) url.
+- Code execution time is limited by the service worker's lifetime which is a few seconds. So abusing device hardware by heavy computation or overusing network resources is rather difficult. For downloading the large files and having their content available for your frontend app, you can use [Background Fetch API](https://docs.microsoft.com/en-us/microsoft-edge/progressive-web-apps-chromium/how-to/background-syncs#use-the-background-fetch-api-to-fetch-large-files-when-the-app-or-service-worker-isnt-running) from the same background services family. This API is designed to keep the user in full control of the long network operation by displaying a browser native UI component with the download progress and Pause/Cancel/Resume buttons.
+- Both `sync` and `periodicsync` events are only fired online. For Background Sync detecting online is a main and only purpose. While for Periodic Background Sync this limitation emphasizes the intended use case: keeping the application itself and/or its data up-to-date by periodically syncing with the server.
```

Comment:
I have a feeling that, given our targeted audience, this is going into a bit too much technical details. It's really great content, and we definitely should talk about this. But I fear that we may be losing folks' attention if we have too much of it. So maybe we should try to summarize the most important points and link to relevant sources instead. What do you think?

Reply:
I agree - this part is a bit longer than I expected and not as engaging as code samples. But it answers two important questions: how do browsers care about readers when they are in PWA user role, and "why it doesn't work" when readers are in developer role. I'd love to give a reference to similar external list but this is something I gathered point by point from multiple sources incl. formal specs myself :)

We can shorten this part a bit for example by removing Bg Fetch "side story"